pkg/devfile/adapters/kubernetes/utils/utils_test.go

```diff
@@ -168,62 +284,86 @@ func TestUpdateContainersWithSupervisord(t *testing.T) {
 								Name: component,
 							},
 						},
+						{
+							Container: &versionsCommon.Container{
+								Name: debugComponent,
+							},
+						},
 					},
 					ExecCommands: tt.execCommands,
 				},
 			}
 
-			containers, err := UpdateContainersWithSupervisord(devObj, tt.containers, tt.runCommand)
+			containers, err := UpdateContainersWithSupervisord(devObj, tt.containers, tt.runCommand, tt.debugCommand, 5858)
 
 			if !tt.wantErr && err != nil {
```

Comment:
This test should be split checking one of the conditions first. Actually, test cases sould be harmonized in the way they check this common pattern. Maybe we should consider extracting this pattern into a helper function?

Reply:
I don't understand what you mean by split checking one of the conditions first.

Reply:
Similar, yes. If you look at the different branches of the if / else, you'll see that several branches test for the same condition (`tt.wantErr` or `err == nil`) and I think it'd be easier to figure out what is actually tested if the conditions were split.

Reply:
Done. I have simplified the error checking logic.